cg/services/bcl_convert_metrics_service/bcl_convert_metrics_service.py

```diff
@@ -0,0 +1,79 @@
+from datetime import datetime
+from pathlib import Path
+
+
+from cg.constants.demultiplexing import UNDETERMINED
+from cg.models.flow_cell.flow_cell import FlowCellDirectoryData
+from cg.services.bcl_convert_metrics_service.parser import MetricsParser
+from cg.store.models import SampleLaneSequencingMetrics
+from cg.utils.flow_cell import get_flow_cell_id
+
+
+class BCLConvertMetricsService:
+    def create_sample_lane_sequencing_metrics_for_flow_cell(
+        self,
+        flow_cell_directory: Path,
+    ) -> list[SampleLaneSequencingMetrics]:
+        """Parse the demultiplexing metrics data into the sequencing statistics model."""
+        metrics_parser = MetricsParser(flow_cell_directory)
+        sample_internal_ids: list[str] = metrics_parser.get_sample_internal_ids()
+        sample_lane_sequencing_metrics: list[SampleLaneSequencingMetrics] = []
+
+        for sample_internal_id in sample_internal_ids:
+            for lane in metrics_parser.get_lanes_for_sample(sample_internal_id=sample_internal_id):
+                metrics: SampleLaneSequencingMetrics = self._create_bcl_convert_sequencing_metrics(
+                    sample_internal_id=sample_internal_id, lane=lane, metrics_parser=metrics_parser
+                )
+                sample_lane_sequencing_metrics.append(metrics)
+        return sample_lane_sequencing_metrics
+
+    def create_undetermined_non_pooled_metrics(
+        self,
+        flow_cell: FlowCellDirectoryData,
+    ) -> list[SampleLaneSequencingMetrics]:
+        """Return sequencing metrics for any undetermined reads in non-pooled lanes."""
+
+        non_pooled_lanes_and_samples: list[tuple[int, str]] = (
+            flow_cell.sample_sheet.get_non_pooled_lanes_and_samples()
+        )
+        metrics_parser = MetricsParser(flow_cell.path)
+        undetermined_metrics: list[SampleLaneSequencingMetrics] = []
+
+        for lane, sample_internal_id in non_pooled_lanes_and_samples:
+            if not metrics_parser.has_undetermined_reads_in_lane(lane):
+                continue
+
+            # Passing Undetermined as the sample id is required to extract the undetermined reads data.
+            # BclConvert tags undetermined reads in a lane with the sample id "Undetermined".
+            metrics: SampleLaneSequencingMetrics = self._create_bcl_convert_sequencing_metrics(
+                sample_internal_id=UNDETERMINED, lane=lane, metrics_parser=metrics_parser
+            )
+            metrics.sample_internal_id = sample_internal_id
+            undetermined_metrics.append(metrics)
+        return undetermined_metrics
+
+    @staticmethod
+    def _create_bcl_convert_sequencing_metrics(
```

Comment:
I think the bcl_convert in this name is something that I forgot to clean. I think it is not necessary as every Illumina flow cell is demultiplexed using BCLConvert

Reply:
we can fix it later, these function perhaps will be gone anyway!